core/core_bind.cpp

```diff
@@ -47,6 +47,7 @@ namespace core_bind {
 ////// ResourceLoader //////
 
 ResourceLoader *ResourceLoader::singleton = nullptr;
+Array ResourceLoader::default_array;
```

Comment:
The default array should probably be read-only to prevent mutating the array, and if we accidentally try to mutate it we'd get an error.

Also, since all these `default_array` instances are the same (empty arrays), instead of having a different instance for each class, we can re-use the same instance everywhere. Maybe we can put it in `Array` (i.e.: `Array::default` or `Array::empty`).

Reply:
Finally took the time to update this PR with your suggestions. I had some difficulties to call `make_read_only` on a static field without static constructors so I added a static method `Array::create_read_only` and used it to create `Array::default`. If you know a better way, I'll be glad to implement it.